src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java

```diff
@@ -0,0 +1,203 @@
+                "activePeopleCycle",
+                "Proportion of adults who cycle at least once per week",
+                "Proportion of adults who cycle at different time periods",
+                "https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/536501/cw0104.ods")
+        ),
+        activePeopleWalk(new DatasourceSpec(
+                ActivePeopleSurveyImporter.class,
+                "activePeopleWalk",
+                "Proportion of adults who walk",
+                "Proportion of adults who walk at different time periods",
+                "https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/536499/cw0105.ods")
+        );
+
+        private DatasourceSpec datasourceSpec;
+        DatasourceId(DatasourceSpec datasourceSpec) {
+            this.datasourceSpec = datasourceSpec;
+        }
+    }
+
+    public ActivePeopleSurveyImporter(){
+        datasourceIds = stringsFromEnumeration(DatasourceId.class);
+    }
+
+    // Instantiating the datasource specifications.
+    @Override
+    public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Exception {
+        return DatasourceId.valueOf(datasourceIdString).datasourceSpec;
+
+    }
+
+    @Override
+    protected List<String> getOaDatasourceIds() {
+        return Arrays.asList(
+                OaImporter.OaType.localAuthority.datasourceSpec.getId(),
+                OaImporter.OaType.englandBoundaries.datasourceSpec.getId());
+    }
+
+    @Override
+    protected void importDatasource(Datasource datasource, List<String> geographyScope, List<String> temporalScope, List<String> datasourceLocation) throws Exception {
+
+        // Check the datasourceid to reference the corresponding url
+        String whichDatasource = datasource.getDatasourceSpec().getId();
+        String fileLocation = getDatasourceSpec(whichDatasource).getUrl();
+        URL url = new URL(fileLocation);
+        InputStream isr = downloadUtils.fetchInputStream(url, getProvider().getLabel(), ".ods");
+
+        SpreadsheetDocument workbook = SpreadsheetDocument.loadDocument(isr);
+        Table sheet = workbook.getSheetByIndex(0);
+        Iterator<Row> rowIterator = sheet.getRowIterator();
+
+        SubjectType localauthority = SubjectTypeUtils.getSubjectTypeByProviderAndLabel(AbstractONSImporter.PROVIDER.getLabel(), getOaDatasourceIds().get(0));
+        SubjectType englandboundaries = SubjectTypeUtils.getSubjectTypeByProviderAndLabel(AbstractONSImporter.PROVIDER.getLabel(), getOaDatasourceIds().get(1));
+
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        int ignore = 0;
+        while (ignore++ < 7) {
+            rowIterator.next();
+        }
+
+        // Getting the indices of the columns containing the data.
+        int columnIndex = 0;
+        List<Integer> columnLoop = new ArrayList<>();
+        while ( columnIndex++ <= 24) {
+            if (!(columnIndex % 5 == 0 && columnIndex != 0)) {
```

Comment:
a negated `and` could be an `or` 🤓

Reply:
yes but you have to flip the conditions as well  `columnIndex % 5 != 0 || columnIndex == 0` 🤓🤓